src/ui/textfield.rs

```diff
@@ -1,15 +1,63 @@
 //! A fairly simplistic text field implementation.
 
+use std::{borrow::BorrowMut, ops::Range};
+
+use copypasta::{ClipboardContext, ClipboardProvider};
 use skulpin::skia_safe::*;
 
 use crate::ui::*;
 
+/// Text field selection struct
+struct Selection {
+    pub first: usize,
+    pub second: usize,
+}
+
+impl Selection {
+    pub fn start(&self) -> usize {
+        self.first.min(self.second)
+    }
+
+    pub fn end(&self) -> usize {
+        self.first.max(self.second)
+    }
+
+    pub fn normalize(&self) -> Range<usize> {
+        self.start()..self.end()
+    }
+
+    pub fn len(&self) -> usize {
+        self.end() - self.start()
+    }
+
+    pub fn move_cursors(&mut self, position: usize) {
```

Comment:
I think a better name for this function would be `move_to`, `move_cursors` implies there are multiple cursors here but in reality there's just one, plus the anchor.